contracts/deployables/counter-sign/CounterSignV1.sol

```diff
@@ -0,0 +1,133 @@
+        (
+            address[] memory _signers,
+            string[] memory _entityNames,
+            uint256 _expiration,
+            string memory _agreementUrl,
+            Transaction[] memory _finalTxs
+        ) = abi.decode(
+                params,
+                (address[], string[], uint256, string, Transaction[])
+            );
+
+        require(
+            _signers.length == _entityNames.length,
+            "Signers and names mismatch"
+        );
+
+        signers = _signers;
+        expirationTimestamp = _expiration;
+        agreementUrl = _agreementUrl;
+        for (uint256 i = 0; i < _finalTxs.length; i++) {
+            finalAgreementTransactions.push(_finalTxs[i]);
+        }
+
+        for (uint256 i = 0; i < _signers.length; i++) {
+            address signer = _signers[i];
+            entityNames[signer] = _entityNames[i];
+        }
+    }
+
+    function getAgreementUrl() external view override returns (string memory) {
+        return agreementUrl;
+    }
+
+    function signAgreement(
+        string memory name
+    ) external override onlyBeforeExpiry {
+        require(_isSigner(msg.sender), "Not a signer");
+        require(!signedAddresses[msg.sender], "Already signed");
+
+        signedAddresses[msg.sender] = true;
+        signerPersonalNames[msg.sender] = name;
+    }
+
+    function executeAgreement() external override {
+        require(getIsExecutable(), "Not all signed");
+        require(!executed, "Already executed");
+
+        for (uint256 i = 0; i < finalAgreementTransactions.length; i++) {
+            _executeTransaction(finalAgreementTransactions[i]);
+        }
+        executed = true;
+    }
+
+    function getSigners() external view override returns (address[] memory) {
+        return signers;
+    }
+
+    function getHasSigned(
+        address signerAddress
+    ) external view override returns (bool) {
+        return signedAddresses[signerAddress];
+    }
+
+    function getIsExecutable() public view override returns (bool) {
+        for (uint256 i = 0; i < signers.length; i++) {
+            if (!signedAddresses[signers[i]]) return false;
+        }
+        return true;
+    }
+
+    function getHasExecuted() external view override returns (bool) {
+        return executed;
+    }
+
+    function getSignerEntityName(
+        address signerAddress
+    ) external view returns (string memory) {
+        return entityNames[signerAddress];
+    }
+
+    function getSignerName(
+        address signerAddress
+    ) external view returns (string memory) {
+        return signerPersonalNames[signerAddress];
+    }
+
+    function _isSigner(address addr) internal view returns (bool) {
+        for (uint256 i = 0; i < signers.length; i++) {
+            if (signers[i] == addr) return true;
+        }
+        return false;
+    }
+
+    function _executeTransaction(Transaction memory txn) internal {
+        bool success;
```

Comment:
I know its redundant (because booleans are initialized as "false" already), but I think it'd be nice to see this as `bool success = false`, considering the "if..else if.." below. If neither of those conditions hit, then we are relying on the `require(success, ...)` to revert the transaction, and so from a readability perspective the "default value of success" needs to be false.